src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -567,6 +569,22 @@ private async Task PackageSourcesChangedAsync(IReadOnlyCollection<PackageSourceC
             }
         }
 
+        private async Task IsCentralPackageManagementEnabledAsync(CancellationToken cancellationToken)
```

Comment:
If I understand correctly then CPM is enabled for all projects or not, so just checking for 1st project would be suffice.

```Suggestion
        private async Task IsCentralPackageManagementEnabledAsync(CancellationToken cancellationToken)
        {
            if (!Model.IsSolution)
            {
                await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
                {
                    // Go off the UI thread to perform non-UI operations
                    await TaskScheduler.Default;
                    if (Model.Context.Projects.Any())
                    {
                        _detailModel.IsCentralPackageManagementEnabled = await Model.Context.Projects.First().IsCentralPackageManagementEnabledAsync(Model.Context.ServiceBroker, cancellationToken);
                    }
                });
            }
        }
```

Reply:
You can skip the `Any()` and just use the `First()` as I suggested here: https://github.com/NuGet/NuGet.Client/pull/5011#discussion_r1103360097

Reply:
But that case we're making assumption, there would be at least 1 project, `Any` is for safeguarding any unexpected null exception error.

Reply:
I'll go with `First()` because as donnie said, we use it on other places and we're not afraid to make that assertion. I would normally go with `Any()` to avoid errors but the PM UI cannot be opened without a project

Reply:
You still didn't implement `First()`. For you no need to loop. You can use above code without `Any()` check too.

Reply:
Sorry I forgot to push new changes.

Reply:
> If I understand correctly then CPM is enabled for all projects or not

That's not correct. CPM (and PackageReference) is implemented via MSBuild, and MSBuild is strictly per-project.  Trying to enforce anything to be "all of solution or none" in MSBuild is effectively impossible.

Reply:
@martinrrm 
Have you looked into above scenario? Do we consider it CPM enabled if some projects are CPM enabled but not all? Maybe check with @jeffkl

Reply:
This code will only run in the project PM UI, so if the user has disabled CPM for that project, it can install any version. I understand that disable CPM in just a project is not a desired behavior for this feature but since its msbuild it difficult to restrain users from doing it. @jeffkl can you confirm this?